lib/model/ui/has_factory.dart

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// This file is separate from has_renderer so it doesn't add in the angular
+// dependency and so 'dart:html' which stops tests from running as a simple
+// dart_test.
```

Comment:
Minor: This comment might need to be revised. I can't quite make out what it is saying (maybe it is just because it is a run-on sentence?).